### DISCUSSION TITLE
osu! lazer has extremely low performance on iPad and other Apple Devices

### DISCUSSION THREAD

**Discussion by nxfx21** [ppy/osu#21328] [Q&A]
### Type

Performance

### Bug description

Osu statters a lot. Especially on iPads. I get 2 fps in game play. (Screenshot in osu tutorial)

### Screenshots or videos

**2 fps on the osu tutorial.**

![9824C8E9-9D4D-4097-9FB6-932EAB74BCBA](https://user-images.githubusercontent.com/75974736/202897744-abb0ee00-fb5f-47ac-b5b0-9029bdca3deb.jpeg)

### Version

osu! 2022.11.0-lazer (and older versions)

### Logs

I don't have iTunes lol

**Comment by bdach** [comment]
Which devices are you testing on exactly?

**Reply by nxfx21** [reply]
iPad 7th generatiom

**Comment by agatemosu** [comment]
Wouldn't the problem be that multithreading is not enabled by default on mobile? When I activated it, the performance on my tablet improved

**Reply by nxfx21** [reply]
It was on while recording

**Comment by frenzibyte** [comment]
osu!(lazer) currently runs on OpenGL backend, which is deprecated technology and likely to perform quite poorly on iOS/iPadOS. Metal support will come very soon (see https://github.com/ppy/osu/issues/18892 / https://github.com/ppy/osu-framework/issues/4909). In addition, performance is a constant work in progress, and mobile platforms are generally not a focus currently, desktop comes first.

Also, the FPS overlay shows that you're already using multi-threaded mode. Can you please reset back to single thread mode and see how that behaves?
